source/slang/slang-check-overload.cpp

```diff
@@ -1395,6 +1398,16 @@ int SemanticsVisitor::CompareLookupResultItems(
             rightIsFreeFormExtension = true;
     }
 
+    // If we are comparing two generic and one of them in an extension, then this generic
```

Comment:
Move this to before 1373?

I wonder if none of these checks should apply to generic decls. Because I can have:

```
extern void foo<T>(int x);
void foo();

void test() { foo<int>(2); } // error??
```